features/apimgt/org.wso2.carbon.apimgt.publisher.feature/src/main/resources/publisher/source/src/app/components/Apis/Details/Environments/Environments.jsx

```diff
@@ -389,10 +399,39 @@ export default function Environments() {
 
     const handleSelect = (event) => {
         const revisions = selectedRevision.filter((r) => r.env !== event.target.name);
-        revisions.push({ env: event.target.name, revision: event.target.value });
+        const theRevision = selectedRevision.find((r) => r.env === event.target.name);
```

Comment:
Shall we refractor this name?

Reply:
fixed with https://github.com/wso2/carbon-apimgt/pull/9913/commits/97d8274d8efaaee56f99da933824466a977a4fc8